#!/usr/bin/env python

PACKAGE = "points_preprocessor_usi"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sensor_height",double_t,   0,
        "sensor height (m)", 2.2,  1.0, 3.0)
gen.add("num_seg",      int_t,      0,
        "useless ", 1, 1, 5)
gen.add("num_iter",     int_t,      0,
        "number iterations", 5, 2, 10)
gen.add("num_lpr",      int_t,      0,
        "num of LPR points (Low Point Representative)", 300, 20, 1000)
gen.add("th_seeds",     double_t,   0,
        "threshold distance of seeds", 1.2, 0.1, 5.0)
gen.add("th_dist",      double_t,   0,
        "distance threhold for ground", 0.3, 0.1, 3.0)

gen.add("th_run",       double_t,   0,
        "distance of points belong to the same run (point to run)",
        0.12, 0.05, 1.0)
gen.add("th_merge", double_t, 0,
        "distance of runs to be merged (run to run)",
        0.5, 0.1, 5.0)


#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                      gen.const("Medium",     int_t, 1, "A medium constant"),
#                      gen.const("Large",      int_t, 2, "A large constant"),
#                      gen.const("ExtraLarge", int_t, 3, "An extra large constant")], "An enum to set size")

#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "dynamic reconfigure", "params"))
